### ISSUE TITLE
`compile_page` fails silently when computed variable raises exception

### ISSUE THREAD

**Issue by masenf** [pynecone-io/pynecone#1277]
**Describe the bug**
If an initial computed var function raises an exception, this isn't reported anywhere, **the page is NOT generated** and the user has no idea what happened or how to identify where the issue is.

**To Reproduce**
```python
import reflex as rx


class State(rx.State):
    @rx.var
    def raises(self) -> str:
        raise RuntimeError("Is this displayed on the console?")


def index() -> rx.Component:
    return rx.heading(
        State.raises,
        padding_top="10%",
    )


app = rx.App(state=State)
app.add_page(index)
app.compile()
```

**Expected behavior**
If there is an error compiling a page, a traceback pointing to the line in question is expected.

** Specifics (please complete the following information):**
 - Python Version: 3.11.4
 - Reflex Version: `main` / `0.2.0a1`
 - OS: macOS 13.4.1

**Comment by masenf** [comment]
More fallout from #1199; when i revert the change. i get the traceback as expected.

**Comment by masenf** [comment]
Fixed by #1281
